ui-admin/src/study/participants/participantList/ParticipantListTable.tsx

```diff
@@ -0,0 +1,228 @@
+  )
+
+  const columns = useMemo<ColumnDef<EnrolleeSearchExpressionResult, string>[]>(() => [{
+    id: 'select',
+    header: ({ table }) => <IndeterminateCheckbox
+      checked={table.getIsAllRowsSelected()} indeterminate={table.getIsSomeRowsSelected()}
+      onChange={table.getToggleAllRowsSelectedHandler()}/>,
+    cell: ({ row }) => (
+      <div className="px-1">
+        <IndeterminateCheckbox
+          checked={row.getIsSelected()} indeterminate={row.getIsSomeSelected()}
+          onChange={row.getToggleSelectedHandler()} disabled={!row.getCanSelect()}/>
+      </div>
+    )
+  }, {
+    header: 'Shortcode',
+    accessorKey: 'enrollee.shortcode',
+    meta: {
+      columnType: 'string'
+    },
+    cell: info => <Link to={`${currentEnvPath}/participants/${info.getValue()}`}>{info.getValue()}</Link>
+  }, {
+    header: 'Created',
+    id: 'createdAt',
+    accessorKey: 'enrollee.createdAt',
+    enableColumnFilter: false,
+    meta: {
+      columnType: 'instant'
+    },
+    cell: info => instantToDefaultString(info.getValue() as unknown as number)
+  }, {
+    id: 'lastLogin',
+    header: 'Last login',
+    enableColumnFilter: false,
+    meta: {
+      columnType: 'instant'
+    },
+    accessorFn: row => row.participantUser?.lastLogin,
+    cell: info => instantToDefaultString(info.getValue() as unknown as number)
+  }, {
+    id: 'familyName',
+    header: 'Family name',
+    accessorFn: row => row.profile?.familyName,
+    meta: {
+      columnType: 'string'
+    }
+  }, {
+    id: 'givenName',
+    header: 'Given name',
+    accessorFn: row => row.profile?.givenName,
+    meta: {
+      columnType: 'string'
+    }
+  }, {
+    id: 'familyShortcode',
+    header: 'Family shortcode',
+    enableColumnFilter: false,
+    meta: {
+      columnType: 'string'
+    },
+    accessorFn: row => row.families.map(family => family.shortcode),
+    cell: ({ row }) => {
+      if (isEmpty(row.original.families)) {
+        return <span className='fst-italic'>None</span>
+      }
+      return <>{row.original.families.map((family, idx) => <div key={idx}><FamilyLink family={family}
+        studyEnvContext={studyEnvContext}/>
+      </div>)}</>
+    }
+  }, {
+    id: 'contactEmail',
+    header: 'Contact email',
+    accessorKey: 'profile.contactEmail',
+    meta: {
+      columnType: 'string'
+    }
+  }, {
+    header: 'Consented',
+    accessorKey: 'enrollee.consented',
+    meta: {
+      columnType: 'boolean',
+      filterOptions: [
+        { value: true, label: 'Consented' },
+        { value: false, label: 'Not Consented' }
+      ]
+    },
+    filterFn: 'equals',
+    cell: info => info.getValue() ? <FontAwesomeIcon icon={faCheck}/> : ''
+  }], [study.shortcode, currentEnv.environmentName])
+
+  const familyLinkageEnabled = studyEnvContext.currentEnv.studyEnvironmentConfig.enableFamilyLinkage
+
+  const table = useReactTable({
+    data: participantList,
+    columns: useMemo(
```

Comment:
this double useMemo was really tricky to follow.  let's move this filtering into the initial column definition useMemo